File: frontend/src/app/dtos/notificationDTO.ts

import { NotificationReferenceType, NotificationType } from './enums';

export interface NotificationDto {
  id: number;
  type: NotificationType;
  message: string;
  referenceId: number | null;
  referenceType: NotificationReferenceType | null;
  isRead: boolean;
  createdAt: string;
}

export interface NotificationSummaryDto {
  unreadCount: number;
  recent: NotificationDto[];
}

export interface AdminNotificationDto {
  id: number;
  userId: string;
  userName: string;
  userAvatarUrl: string | null;
  type: NotificationType;
  message: string;
  referenceId: number | null;
  referenceType: NotificationReferenceType | null;
  isRead: boolean;
  createdAt: string;
}

export interface MarkMultipleNotificationsReadDto {
  notificationIds: number[];
}

export interface UnreadNotificationCountDto {
  unreadCount: number;
}

File: backend/Dtos/NotificationDto.cs
using backend.Models;
using System.ComponentModel.DataAnnotations;

namespace backend.Dtos
{
    //list?
    public class NotificationSummaryDto
    {
        public int UnreadCount { get; set; }
        public List<NotificationDto> Recent { get; set; } = new();
    }
    public class NotificationDto
    {
        public int Id { get; set; }

        public NotificationType Type { get; set; }
        public string Message { get; set; } = string.Empty;

        public int? ReferenceId { get; set; }
        public NotificationReferenceType? ReferenceType { get; set; }
        
        public bool IsRead { get; set; }
        public DateTime CreatedAt { get; set; }

    }

    public class AdminNotificationDto
    {
        public int Id { get; set; }

        //Receiver info (for admin view)
        public string UserId { get; set; } = string.Empty;
        public string UserName { get; set; } = string.Empty;
        public string? UserAvatarUrl { get; set; }

        //Notification details
        public NotificationType Type { get; set; }
        public string Message { get; set; } = string.Empty;

        //Related entity
        public int? ReferenceId { get; set; }
        public NotificationReferenceType? ReferenceType { get; set; }

        public bool IsRead { get; set; }

        public DateTime CreatedAt { get; set; }
    }

    public class MarkMultipleNotificationsReadDto
    {
        [Required(ErrorMessage = "Notification IDs are required")]
        public List<int> NotificationIds { get; set; } = new();
    }

    public class UnreadNotificationCountDto
    {
        public int UnreadCount { get; set; }
    }

  

   

}